libpromises/evalfunction.c

```diff
@@ -8450,6 +8451,84 @@ static FnCallResult FnCallNetworkConnections(EvalContext *ctx, ARG_UNUSED const
 
 /*********************************************************************/
 
+static FnCallResult FnCallFileReadable(ARG_UNUSED EvalContext *const ctx,
+                                       ARG_UNUSED const Policy *const policy,
+                                       const FnCall *const fp,
+                                       const Rlist *finalargs)
+{
+    assert(fp != NULL);
+    assert(fp->name != NULL);
+
+    if (finalargs == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Function %s requires path as first argument",
+            fp->name);
+        return FnFailure();
+    }
+    const char *const path = RlistScalarValue(finalargs);
+
+    finalargs = finalargs->next;
+    const long timeout = (finalargs == NULL) ? 0L
+                       : IntFromString(RlistScalarValue(finalargs));
+    assert(timeout >= 0);
```

Comment:
What happens if you specify a negative value in policy?

Reply:
Then you will get the following error message:
```
   error: Fatal CFEngine error: In function 'filereadable', error in variable '-1', 'Integer is out of range'
```

Reply:
Great!

Reply:
Also if the time limit for some reason would be negative, `select` would return `-1` and set `errno = EINVAL`